### DISCUSSION TITLE
Running data driven tests concurrently

### DISCUSSION THREAD

**Discussion by tday** [vitest-dev/vitest#4051] [Q&A] [answered]
Hello, I have a simple data driven test structure like so:

```
describe.concurrent.each([scenario1, scenario2])('given scenario %s', () => {
   it('should not fail', () => {
     expect(true).toBeTruthy();
  });
});
```

I was expecting that it would run scenario1 and scenario2 to run concurrently. However, it runs them sequentially. It seems it only runs a second test within the describe concurrently. 

Is there a way to run the data driven tests concurrently as though they were multiple tests within the describe?

**Comment by sheremet-va** [comment]
> It seems it only runs a second test within the describe concurrently.

Your code generates two suites with a single test inside. What do you mean by a second test?

**Reply by tday** [reply]
Sorry, I should have provided a second example.

a and b will run concurrently for each test scenario. Each test scenario runs sequentially.

```
describe.concurrent.each([scenario1, scenario2])('given scenario %s', () => {
   it('a', () => {
     expect(true).toBeTruthy();
  });

  it('b, () => {
    expect(true).toBeTruthy();
  });
});
```

**Reply by tday** [reply]
Is there a way to get the scenarios to run concurrently?

**Reply by sheremet-va** [reply]
So, you are saying that there are 4 tests, and the first 2 are running concurrently, and then the other 2?

**Reply by tday** [reply]
Yep! That's how it appears to me watching the console logs.

**Reply by tday** [reply]
https://github.com/vitest-dev/vitest/assets/13629388/ee13bcca-6305-408a-9023-e2c1c1f58c7c

**Reply by sheremet-va** [reply, marked as answer]
Concurrent tests are running concurrently only inside a single suite in Vitest. This sounds like a nice enhancement feature though.

**Reply by tday** [reply]
Got it. Thanks for verifying that I wasn't doing it wrong.
